File: BusInfo/app/src/main/java/com/daeseong/businfo/MainActivity.java
package com.daeseong.businfo;

import androidx.appcompat.app.AppCompatActivity;
import android.content.pm.ActivityInfo;
import android.graphics.Color;
import android.os.Build;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;
import android.view.inputmethod.InputMethodManager;
import com.google.android.material.tabs.TabLayout;

public class MainActivity extends AppCompatActivity {

    private static final String TAG = MainActivity.class.getSimpleName();

    private View include;
    private TabLayout mainTabLayout;
    private SwipeViewPager mainViewPager;
    private MainPagerAdapter mainPagerAdapter = null;
    private int currentTabIndex = 0;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        InitTitleBar();

        setContentView(R.layout.activity_main);

        init();

        setViewPager();
        setInitTabLayout();
    }

    private void InitTitleBar(){

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            Window window = getWindow();
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
            window.setStatusBarColor(Color.rgb(255, 255, 255));
        }

        try {
            //안드로이드 8.0 오레오 버전에서만 오류 발생
            setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_PORTRAIT);
            supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        } catch (Exception ex) {
        }
    }

    private void init(){

        mainTabLayout = findViewById(R.id.maintabLayout);
        mainViewPager = findViewById(R.id.mainviewPager);
        include = findViewById(R.id.include_maintoolbar);

        View cLbusmenu = include.findViewById(R.id.cLbusmenu);
        cLbusmenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                try {
                    Log.e(TAG, "상단 메뉴 클릭");
                } catch (Exception ex) {
                }
            }
        });
    }

    private void setViewPager(){
        mainPagerAdapter = new MainPagerAdapter(getSupportFragmentManager());
        mainViewPager.setAdapter(mainPagerAdapter);
        mainTabLayout.setupWithViewPager(mainViewPager);
    }

    private void setInitTabLayout(){

        mainTabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                try {
                    currentTabIndex = tab.getPosition();
                    mainViewPager.setCurrentItem(currentTabIndex);
                } catch (Exception ex) {
                }
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {
            }
        });

        SwipeViewPager.OnPageChangeListener viewPagerPageChangeListener = new SwipeViewPager.OnPageChangeListener() {
            @Override
            public void onPageSelected(int position) {
                Log.e(TAG, "onPageSelected:" + position);
            }

            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }

            @Override
            public void onPageScrollStateChanged(int state) {
            }
        };
        mainViewPager.addOnPageChangeListener(viewPagerPageChangeListener);
    }

    public void hideKeyboard() {

        try {
            if (getCurrentFocus() != null) {
                InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(INPUT_METHOD_SERVICE);
                inputMethodManager.hideSoftInputFromWindow(getCurrentFocus().getWindowToken(), 0);
            }
        } catch (Exception ex) {
        }
    }
}

File: BusInfo(kotlin)/app/src/main/java/com/daeseong/businfo/MainActivity.kt
package com.daeseong.businfo

import android.content.pm.ActivityInfo
import android.graphics.Color
import android.os.Build
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.Window
import android.view.WindowManager
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.app.AppCompatActivity
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.google.android.material.tabs.TabLayout

class MainActivity : AppCompatActivity() {

    private val tag = MainActivity::class.java.simpleName

    private lateinit var include: View
    private lateinit var mainTabLayout: TabLayout
    private lateinit var mainViewPager: SwipeViewPager
    private var mainPagerAdapter: MainPagerAdapter? = null
    private var currentTabIndex = 0

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        initTitleBar()

        setContentView(R.layout.activity_main)

        init()

        setViewPager()
        setInitTabLayout()
    }

    private fun initTitleBar() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            val window: Window = window
            window.addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS)
            window.statusBarColor = Color.rgb(255, 255, 255)
        }

        try {
            // 안드로이드 8.0 오레오 버전에서만 오류 발생
            requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
            supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        } catch (ex: Exception) {
        }
    }

    private fun init() {
        mainTabLayout = findViewById(R.id.maintabLayout)
        mainViewPager = findViewById(R.id.mainviewPager)
        include = findViewById(R.id.include_maintoolbar)

        val cLbusmenu: View = include.findViewById(R.id.cLbusmenu)
        cLbusmenu.setOnClickListener {
            try {
                Log.e(tag, "상단 메뉴 클릭")
            } catch (ex: Exception) {
            }
        }
    }

    private fun setViewPager() {
        mainPagerAdapter = MainPagerAdapter(supportFragmentManager)
        mainViewPager.adapter = mainPagerAdapter
        mainTabLayout.setupWithViewPager(mainViewPager)
    }

    private fun setInitTabLayout() {
        mainTabLayout.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab) {
                try {
                    currentTabIndex = tab.position
                    mainViewPager.currentItem = currentTabIndex
                } catch (ex: Exception) {
                }
            }

            override fun onTabUnselected(tab: TabLayout.Tab) {}

            override fun onTabReselected(tab: TabLayout.Tab) {}
        })

        val viewPagerPageChangeListener: OnPageChangeListener = object : OnPageChangeListener {
            override fun onPageSelected(position: Int) {
                Log.e(tag, "onPageSelected:$position")
            }

            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
            }

            override fun onPageScrollStateChanged(state: Int) {}
        }
        mainViewPager.addOnPageChangeListener(viewPagerPageChangeListener)
    }

    fun hideKeyboard() {
        try {
            currentFocus?.let {
                val inputMethodManager = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                inputMethodManager.hideSoftInputFromWindow(it.windowToken, 0)
            }
        } catch (ex: Exception) {
        }
    }
}
